internal/services/managedhsm/parse/nested_item.go

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package parse
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/resourceids"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+// NestedItemObjectType enumerates the type of the "NestedItemType" value (e.g."keys", "secrets" or "certificates").
+type NestedItemObjectType string
+
+const (
+	// KeyVaultObjectType Keys...
+	NestedItemTypeKey NestedItemObjectType = "keys"
+)
```

Comment:
We're intending to move away from the `NestedItemId` in favour of Resource ID types specific to the type of Nested Item (e.g. a `ManagedHSMKey` etc) - as such can we update this Resource ID type to be one for each type?

Reply:
This is still pending FWIW